content-services/content-sources-backend
# Fixes 3602: api to check presence of packages in repos

## Summary

- Adds api to check if one or more package names exist in a set of repositories (given by URL and/or UUID)

## Testing steps

- Create/introspect a few repositories
- Check for package names that exist in those repos with this endpoint and sample request: 

`POST /rpms/presence`

```
{
  "rpm_names": ["giraffe", "lion", "elephant", "6tunnel", "zebra"],
  "urls": [
    "https://fedorapeople.org/groups/katello/fakerepos/zoo/",
    "https://fedorapeople.org/groups/katello/fakerepos/zoo2/"
  ],
  "uuids": [ "5fc43beb-975f-44c2-98b0-584640d7ea55"
  ]
}
```

Response should return found and missing packages:

```
{
    "found": [
        "6tunnel",
        "elephant",
        "giraffe",
        "lion"
    ],
    "missing": [
        "zebra"
    ]
}
```

## Checklist
